src/neo/SmartContract/Manifest/ContractAbi.cs

```diff
@@ -1,20 +1,39 @@
+using Neo.IO;
 using Neo.IO.Json;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 
 namespace Neo.SmartContract.Manifest
 {
     /// <summary>
     /// For technical details of ABI, please refer to NEP-3: NeoContract ABI. (https://github.com/neo-project/proposals/blob/master/nep-3.mediawiki)
     /// </summary>
-    public class ContractAbi
+    public class ContractAbi : ISerializable
     {
+        /// <summary>
+        /// Max length for a valid Contract Abi
+        /// </summary>
+        public const int MaxLength = 4096;
+
         private IReadOnlyDictionary<string, ContractMethodDescriptor> methodDictionary;
 
         /// <summary>
-        /// Hash is the script hash of the contract. It is encoded as a hexadecimal string in big-endian.
+        /// Serialized size
+        /// </summary>
+        public int Size
+        {
+            get
+            {
+                int size = Length;
+                return IO.Helper.GetVarSize(size) + size;
+            }
+        }
```

Comment:
Why do we need both `Length` and `Size`?

Reply:
`Size` it's required because it's `ISerializable` and we use `Length` for check the length against `ContractAbi.MaxLength`

Reply:
The `Size` should be the `Length`.

Reply:
Done